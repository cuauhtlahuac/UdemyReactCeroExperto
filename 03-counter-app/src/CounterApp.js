import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
	const [ counter, setCounter ] = useState(value);

	const handleAdd = (e) => {
		// setCounter(counter + 1);
		setCounter((c) => c + 1); // el argumento es el counter state!!!
	};
	const handleSub = (e) => {
		// setCounter(counter + 1);
		setCounter((c) => c - 1); // el argumento es el counter state!!!
	};

	return (
		<div>
			<h1>CounterApp</h1>
			<h2> {counter} </h2>
			<button onClick={handleAdd}>↑1</button>
			<button onClick={() => setCounter(value)}>reset</button>
			<button onClick={handleSub}>↓1</button>
		</div>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number
};

export default CounterApp;
