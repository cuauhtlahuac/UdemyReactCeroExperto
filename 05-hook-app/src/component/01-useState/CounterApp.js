import React, { useState } from 'react'

export const CounterApp = () => {
    const [state, setState] = 
    useState(
        { counter1: 10,
          counter2: 20,
          counter3: 30,
          counter4: 40,
        })
    
    const { counter1, counter2 } = state;
        
    return (
        <>
         <h1>Counter 1 = { counter1 }</h1>
         <h1>Counter 2 = { counter2 }</h1>
         <hr />

         <button
          className='btn btn-primary'
          onClick={ 
              () => setState({
                    ...state,
                    counter1: counter1 + 1 
                    })
                }
          > +1 </button>
        </>
    )
}
