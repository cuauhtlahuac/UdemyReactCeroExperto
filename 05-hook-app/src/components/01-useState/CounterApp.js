import React, { useState } from 'react'

export const CounterApp = () => {
    const [state, setState] = 
    useState(
        { counter1: 10,
          counter2: 20,
          counter3: 30,
          counter4: 40,
        })
    
    const { counter1, counter2 } = state;
        
    return (
        <>
         <h2>Counter 1 = { counter1 }</h2>
         <h2>Counter 2 = { counter2 }</h2>
         <hr />

         <button
          className='btn btn-primary'
          onClick={ 
              () => setState({
                    ...state,
                    counter1: counter1 + 1 
                    })
                }
          > +1 </button>
        </>
    )
}
