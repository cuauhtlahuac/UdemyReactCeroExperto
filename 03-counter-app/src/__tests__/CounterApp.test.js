import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import toJSON from 'enzyme-to-json';

import CounterApp from 'CounterApp';

describe('<CounterApp /> tests', () => {
	test('should render <CounterApp /> Correctly', () => {
		const component = shallow(<CounterApp />);

		expect(component).toMatchSnapshot();
	});

	test('should render 100 as value', () => {
		const value = 100;
		const component = shallow(<CounterApp value={value} />);
		const h2 = component.find('h2').text();

		expect(h2).toBe(` ${value} `);

		expect(toJSON(component)).toMatchSnapshot();
	});
});
