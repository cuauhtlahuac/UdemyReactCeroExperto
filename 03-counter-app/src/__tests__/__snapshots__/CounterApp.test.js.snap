// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> tests should render <CounterApp /> Correctly 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    ↑1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    ↓1
  </button>
</div>
`;

exports[`<CounterApp /> tests should render 100 as value 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    onClick={[Function]}
  >
    ↑1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    ↓1
  </button>
</div>
`;
